.github/workflows/pr.yml

```diff
@@ -31,76 +24,77 @@ jobs:
       - name: Validate ktlint
         run: ./gradlew ktlintCheck
 
+      -   name: Upload lint reports (SARIF)
+          if: ${{ !cancelled() && hashFiles('**/*.sarif') != '' }}
+          uses: github/codeql-action/upload-sarif@v3
+          with:
+              sarif_file: './'
+
   lint:
     runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
```

Comment:
Why is this needed? I do not see it mentioned on https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/uploading-a-sarif-file-to-github.

Reply:
Good point it's a mistake.

Reply:
I do remember why I added this permission. If you look in the PR comments Github post a comment. I initially though it needed this permission to push a comment on the PR. I did remove the permission and I'm gonna check in another PR if it works properly without the permission.